pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -766,11 +711,15 @@ func TestGetTaskFunc_V1beta1Task_VerifyNoError(t *testing.T) {
 	tektonclient := fake.NewSimpleClientset()
 
 	unsignedTask := test.GetUnsignedTask("test-task")
+	unsignedV1Task := &v1.Task{}
+	unsignedTask.ConvertTo(ctx, unsignedV1Task)
+	unsignedV1Task.APIVersion = "tekton.dev/v1"
+	unsignedV1Task.Kind = "Task"
```

Comment:
Just want to double check-- I think you might have forgotten to rebase these changes but we already have tests for v1 tasks, so this shouldn't be necessary.

Reply:
My understanding is that the conversion here happened for the only purpose of comparing `tc.expected` with the resolved Task since with v1 storage version the resolved Task would also be in v1. And the remote task being tested with the GetTaskFunc is `v1beta1`, which put into the mock requester. 

I am thinking what would be a nice way to refactor these tests if the above makes sense: - probably also rename `getUnsignedTask` to `getV1beta1UnsignedTask` or shall I also add a helper function here for converting v1beta1 unsigned Task to v1 named `convertV1beta1ReferencedTaskForComparing`?

Reply:
Opened https://github.com/tektoncd/pipeline/pull/6814 to mitigate the confusions as above.

Reply:
oh I see. what I think would be preferable is to allow `GetUnsignedTask/Pipeline` to accept an api version, rather than calling conversion logic in the test. this would reduce the amount of logic in the test, so if the test fails we can be more confident the problem is with the taskref code rather than being unsure if it's in the taskref code or conversion code.

Reply:
Opened https://github.com/tektoncd/pipeline/pull/6815 to track this, I think we could mark this as resolved and refactor within the scope of #6815 . Feel free to unresolve if this is considered as a blocker of the PR